services/app-api/src/authn/thirdPartyAuthn.ts

```diff
@@ -17,8 +18,20 @@ export async function userFromThirdPartyAuthorizer(
 
     const serviceName = 'third-party-authorizer'
     initTracer(serviceName, otelCollectorURL)
-
+    const allowedIpAddresses = process.env.ALLOWED_IP_ADDRESSES
```

Comment:
Please don't read env vars outside of the initialization of one of our services. I try for us to do all that kind of checking and throwing at the apollo_gql.ts level and then have everything else take parameters. This makes things easier to test and ensures that everything is configured at startup, and gives us nicer interfaces to things with types ensuring that everything is present. Happy to talk about this more this is one of my big architecture goals for the app.

Reply:
👍 I would be interested in hearing more on this